src/app/pages/send-tokens/components/recipient-field.tsx

```diff
@@ -25,22 +48,65 @@ function RecipientFieldBase(props: RecipientField) {
           mb="tight"
           fontSize={1}
           fontWeight="500"
-          htmlFor="recipient"
+          htmlFor="recipientAddressOrBnsName"
         >
           Recipient
         </Text>
         <Input
           display="block"
           type="string"
           width="100%"
-          name="recipient"
+          name="recipientAddressOrBnsName"
           value={value}
           onChange={handleChange}
-          placeholder="Enter an address"
+          onBlur={async () => {
+            setResolvedBnsAddress('');
+            try {
+              const res = await client.namesApi.getNameInfo({ name: value ?? '' });
+              if (res.address) {
+                setFieldValue('recipient', res.address);
+                setResolvedBnsAddress(res.address);
+              } else {
+                setFieldValue('recipient', values.recipientAddressOrBnsName);
+              }
+              return;
+            } catch {
+              // do nothing
```

Comment:
Is this the best way to leave this?

Reply:
I'm not sure, didn't want to leave the block empty to avoid linter errors. Was considering setting the value of the field here rather than outside the try-catch blocks. Since it's not strictly necessary to have it outside, I'll add it here so this looks cleaner.